doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java

```diff
@@ -159,6 +134,21 @@ public void updateDoiRequest(UUID publicationIdentifier, DoiRequestStatus reques
         putItem(publication);
     }
 
+    private List<Publication> extractMostRecentVersionOfEachPublication(URI publisher) throws DynamoDBException {
+        return attempt(() -> queryByPublisher(publisher))
+            .map(this::extractPublications)
+            .map(this::keepMostRecentPublications)
+            .orElseThrow(this::handleErrorFetchingPublications);
+    }
+
+    private boolean hasDoiRequestStatus(Publication publication, DoiRequestStatus desiredStatus) {
```

Comment:
I am not sure if the Publication class should have a method for every case that we cover in any service. It seems that Publication does not have other methods except for getters and setters.

Reply:
Yes, I know. I am open to suggestions but this might be a step in direction of having multiple ways of handling things. Maybe we need a service or wrapper or something.

Reply:
Do you me to create the wrapper in scope of this PR?

Reply:
I agree about thoughts @janbj mentions, that we want this as methods on the business object of Publication, but atm this is the Datamodel which is more of a pure DAO. So we are probably interesting in as quickly as possible to rename the model classes to be DAOs and then make the Publication business object to eat the datamodel DAO. 

Might be out of scope, but problem is that this is something we really should devote time to fix because it just gets uglier and uglier.

Reply:
ie I have code in the data model where Ive started using it as business object due to the mantra of telling objects what to do, instead of asking em about information.